### PR TITLE
Cleanup headers in irr/include/

### PR DESCRIPTION
**Pull request by Desour**
* Goal: Headers in irr/include/ are included even in src/, so we want them to not needlessly include more than necessary. (Stuff in irr/src/ doesn't matter as much for compile time.) (Headers in src/ may be tackled in another PR.)
* What I did:
  * I sorted the includes topologically (with <https://github.com/bkryza/clang-include-graph>), so I can start from the leafs. (Otherwise, changes in later changed files could've affected earlier ones.)
  * I opened the files in order with an IDE with language server support, and let clangd (which has <https://clangd.llvm.org/design/include-cleaner>) tell me about unnecessary includes.
  * I've removed unnecessary includes and added IWYU pragmas (<https://github.com/include-what-you-use/include-what-you-use/blob/master/docs/IWYUPragmas.md>) as necessary and sensible, to suppress warnings.
  * If there were other warnings, I've also fixed them.
  * Then I've fixed the includers until minetest compiled again.

## To do

This PR is a Ready for Review.

## How to test

It builds.

### CODE REVIEW

**Comment by sfan5** [comment]
I think we should merge #15126 before any big cleanups, to avoid everything turning into a mess.

**Comment by Desour** [comment]
Rebased.

File: irr/scripts/BindingGenerator.lua [hunk 1]
```diff
@@ -198,7 +198,7 @@ local consts = List();
 -- Parse a whole header, extracting the data.
 local function ParseHeader( path, into, apiRegex, defs, consts, nameSet, noNewNames )
 	defs:AddFormat( "\t// %s", path );
-	local f = assert( io.open( path, "r" ), "Could not open " .. path );
```
Comment (SmallJoker) [inline irr/scripts/BindingGenerator.lua:201]: Rebase error?

File: src/irrlicht_changes/CGUITTFont.cpp [hunk 2]
```diff
@@ -30,9 +30,15 @@
...
-#include <irrlicht.h>
 #include <iostream>
 #include "CGUITTFont.h"
+#include "CMeshBuffer.h"
+#include "IFileSystem.h"
+#include "IGUIEnvironment.h"
+#include "IMeshManipulator.h"
+#include "ISceneNode.h"
+#include "IMeshSceneNode.h"
+#include "ISceneManager.h"
```
Comment (SmallJoker) [inline src/irrlicht_changes/CGUITTFont.cpp:41]: Optional goal: Alphabetical order to find things faster.

**Review by SmallJoker — COMMENTED** [review]
Looks good otherwise. I appreciate this cleanup a lot.

File: src/irrlicht_changes/CGUITTFont.cpp (hunk 2 above)
Comment (Desour) [reply src/irrlicht_changes/CGUITTFont.cpp:41]: Won't do it everywhere though, to not blow up changed lines.

**Comment by nerzhul** [comment]
great job, that should decrease a bit compilation time

**Review by nerzhul — APPROVED** [review]

**Comment by Desour** [comment]
If someone wants to continue on this before I do:
* Here's the topological sort for files in src/: [toposort_src.txt](https://github.com/user-attachments/files/17435071/toposort_src.txt)
  Made with: `clang-include-graph -r <path_to_luanti>/src/ -l -s > toposort_src.txt`
* Start at top.
* I suggest not doing all in one PR. This PR also only had 118 files, and many of them were trivial. Smaller PRs are easier to review.
* Please write down if you're working on it.
* `irrlichttypes_bloated.h` and `irrlichttypes_bloated.h` should probably be removed.
* See top comment for more instructions.
* Thanks!

**Pull request merged**